connectors/source.py

```diff
@@ -20,6 +20,31 @@
 )
 
 
+class ConfigurationFieldMissingError(ValueError):
```

Comment:
Why different exception types?

I don't see any code catching these types, so it might be an overkill?

Reply:
I thought it's useful to keep them distinct from each other, especially if we provide specific constructors, e.g. as you've pointed out we can pass the missing fields to `ConfigurationFieldMissingError` and for `ConfigurationFieldTypeError ` we could pass the expected type, actual type, field and value.

Reply:
Yeah makes sense, I think we can extend these error classes later with specific constructors, awesome!